jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/internal/ContentSourceByteBuffer.java

```
@@ -47,6 +47,8 @@ public void run()
             if (Content.Chunk.isFailure(chunk))
             {
                 promise.failed(chunk.getFailure());
+                if (!chunk.isLast())
+                    source.fail(chunk.getFailure());
```

This is reading and accumulating into a single BB.
If we have a transient failure, should we not just continue as if we read `null`?

---

That would silently ignore idle timeouts, which I think isn't a reasonable default. We could eventually add a mechanism that allows taking decisions on what to do with failures, but that's outside the scope of this PR.